share/cmake/modules/Findpybind11.cmake

```diff
@@ -22,28 +23,88 @@ endif()
 ### Try to find package ###
 
 if(NOT OCIO_INSTALL_EXT_PACKAGES STREQUAL ALL)
-    set(_pybind11_SEARCH_DIRS
-        ${pybind11_DIRS}
-        ~/Library/Frameworks
-        /Library/Frameworks
-        /usr/local
-        /usr
-        /sw        # Fink
-        /opt/local # DarwinPorts
-        /opt/csw   # Blastwave
-        /opt
-    )
+    if(DEFINED pybind11_DIRS)
+        # Try user defined search path. Ignore all default cmake and system 
+        # paths, and fall back on a system CMake config.
+        find_path(pybind11_INCLUDE_DIR
+            NAMES
+                pybind11/pybind11.h
+            HINTS
+                ${pybind11_DIRS}
+            PATH_SUFFIXES
+                include
+                pybind11/include
+            NO_DEFAULT_PATH
+        )
 
-    # Find include directory
-    find_path(pybind11_INCLUDE_DIR
-        NAMES
-            pybind11/pybind11.h
-        HINTS
-            ${_pybind11_SEARCH_DIRS}
-        PATH_SUFFIXES
-            include
-            pybind11/include
-    )
+        if(EXISTS "${pybind11_INCLUDE_DIR}")
+            # Assume user-provided version is OK.
+            # TODO: Can the pybind11 version be inspected from a header?
+            set(pybind11_VERSION ${pybind11_FIND_VERSION})
+            message(STATUS 
+                "Using user defined search path for pybind11 "
+                "(version \">=${pybind11_FIND_VERSION}\" assumed)"
+            )
+        endif()
+    endif()
+
+    if(NOT DEFINED pybind11_VERSION)
+        # Search for a CMake config, which should have been installed with 
+        # pybind11, and will provide version information.
+        find_package(pybind11 ${pybind11_FIND_VERSION} CONFIG)
+
+        if(NOT pybind11_FOUND)
```

Comment:
Even if an installed version is found, the version must still be validated.

Reply:
`find_package_handle_standard_args` will validate the version based on the `find_package` call that resolved to run this module.